res/controllers/Roland_DJ-202-scripts.js

```diff
@@ -0,0 +1,1612 @@
+DJ202.cueLoopLength = 2;
+DJ202.slicerBeatsWindow = 8;
+DJ202.autoFocusEffects = false;
+DJ202.autoShowFourDecks = false;
+DJ202.bindSamplerControls = false;
+DJ202.vinylModeEnabledOnStart = true;
+
+
+///////////
+// Code. //
+///////////
+
+DJ202.PadMode = {
+    HOTCUE: 1,
+    CUELOOP: 2,
+    PITCHPLAY: 3,
+
+    LOOP: 4,
+    ROLL: 5,
+
+    SEQUENCER: 6,
+    // mode: Use Pads 1-8 to play your samples in time with the sequencer.
+    INSTRUMENTRECORD: 7,
+    PATTERN: 8,
+
+    SAMPLER: 9,
+    SLICERLOOP: 10,
+    SLICER: 11,
+};
+
+DJ202.pitchplayRange = {
+    UP: 0,
+    MID: 1,
+    DOWN: 2,
+};
+
+DJ202.init = function() {
+    DJ202.shiftButton = function(_channel, _control, value, _status, _group) {
+        DJ202.deck.concat(DJ202.effectUnit, DJ202.sampler, DJ202.browseEncoder)
+            .forEach(value ? function(module) {
+                module.shift();
+            } :
+                function(module) {
+                    module.unshift();
+                });
+    };
+
+    DJ202.leftDeck = new DJ202.Deck([1, 3], 0);
+    DJ202.rightDeck = new DJ202.Deck([2, 4], 1);
+    DJ202.deck = [DJ202.leftDeck, DJ202.rightDeck];
+
+    DJ202.sampler = new DJ202.Sampler();
+    DJ202.sampler.reconnectComponents();
+
+    DJ202.effectUnit = [];
+    DJ202.effectUnit[1] = new DJ202.EffectUnit(1);
+    DJ202.effectUnit[2] = new DJ202.EffectUnit(2);
+
+    engine.makeConnection("[Channel3]", "track_loaded", DJ202.autoShowDecks);
+    engine.makeConnection("[Channel4]", "track_loaded", DJ202.autoShowDecks);
+
+    if (engine.getValue("[Master]", "num_samplers") < 16) {
+        engine.setValue("[Master]", "num_samplers", 16);
+    }
+
+    // request initial state
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x00, 0xF7], 6);
+    // unlock pad layers
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x01, 0xF7], 6);
+
+    // engine.beginTimer(500, function() {
+    //   // Keep sending this message to enable performance pad LEDs
+    //   // on the DJ 505, not sure this is neerded for the 202
+    //   midi.sendShortMsg(0xBF, 0x64, 0x00);
+    // });
+
+    DJ202.leftDeck.setCurrentDeck("[Channel1]");
+    DJ202.rightDeck.setCurrentDeck("[Channel2]");
+
+    // Initialize vinyl led state
+    DJ202.leftDeck.setVinylModeLED();
+    DJ202.rightDeck.setVinylModeLED();
+};
+
+DJ202.autoShowDecks = function(_value, _group, _control) {
+    const anyLoaded = engine.getValue("[Channel3]", "track_loaded") ||
+    engine.getValue("[Channel4]", "track_loaded");
+    if (!DJ202.autoShowFourDecks) {
+        return;
+    }
+    engine.setValue("[Master]", "show_4decks", anyLoaded);
+};
+
+DJ202.shutdown = function() {};
+
+DJ202.sortLibrary = function(_channel, control, value, _status, _group) {
+    if (value === 0) {
+        return;
+    }
+
+    let sortColumn;
+    switch (control) {
+    case 0x12: // SONG
+        sortColumn = 1;
+        break;
+    case 0x13: // BPM
+        sortColumn = 14;
+        break;
```

Comment:
```suggestion
    case 0x12: // SONG
        sortColumn = script.LIBRARY_COLUMNS.ARTIST;
        break;
    case 0x13: // BPM
        sortColumn = script.LIBRARY_COLUMNS.BPM;
        break;
```
These constants are defined here: 
https://github.com/mixxxdj/mixxx/blob/e2d41999e7c017c6a353f87967152d3bd2102354/res/controllers/common-controller-scripts.js#L133-L159